crates/fuel-core/src/service/adapters/consensus_module/poa/pre_confirmation_signature/broadcast.rs

```diff
@@ -1,42 +1,144 @@
-use crate::service::adapters::consensus_module::poa::pre_confirmation_signature::{
-    key_generator::Ed25519Key,
-    parent_signature::FuelParentSigner,
+use crate::service::adapters::{
+    consensus_module::poa::pre_confirmation_signature::{
+        key_generator::Ed25519Key,
+        parent_signature::FuelParentSigner,
+    },
+    P2PAdapter,
 };
 use fuel_core_poa::pre_confirmation_signature_service::{
     broadcast::{
         Broadcast,
         PublicKey,
-        Signature,
     },
-    error::Result as PoAResult,
+    error::{
+        Error as PreConfServiceError,
+        Result as PreConfServiceResult,
+    },
     parent_signature::ParentSignature,
 };
-use fuel_core_types::services::p2p::{
-    DelegatePreConfirmationKey,
-    Preconfirmation,
+use fuel_core_types::{
+    services::p2p::{
+        DelegatePreConfirmationKey,
+        PreConfirmationMessage,
+        Preconfirmation,
+        Preconfirmations,
+        SignedPreconfirmationByDelegate,
+    },
+    tai64::Tai64,
 };
 
-/// TODO: Implement `Broadcast` properly: <https://github.com/FuelLabs/fuel-core/issues/2783>
-pub struct P2PBroadcast;
+use fuel_core_poa::pre_confirmation_signature_service::signing_key::SigningKey;
+use fuel_core_types::fuel_crypto::Signature;
+use std::sync::Arc;
 
-impl Broadcast for P2PBroadcast {
-    type DelegateKey = Ed25519Key;
+impl Broadcast for P2PAdapter {
     type ParentKey = FuelParentSigner;
+    type DelegateKey = Ed25519Key;
     type Preconfirmations = Vec<Preconfirmation>;
 
     async fn broadcast_preconfirmations(
         &mut self,
-        _: Self::Preconfirmations,
-        _: Signature<Self>,
-    ) -> PoAResult<()> {
-        todo!()
+        preconfirmations: Self::Preconfirmations,
+        signature: <Self::DelegateKey as SigningKey>::Signature,
+        expiration: Tai64,
+    ) -> PreConfServiceResult<()> {
+        if let Some(p2p) = &self.service {
+            let entity = Preconfirmations {
+                expiration,
+                preconfirmations,
+            };
+            let signature_bytes = signature.to_bytes();
+            let signature = Signature::from_bytes(signature_bytes);
+            let preconfirmations = Arc::new(PreConfirmationMessage::Preconfirmations(
+                SignedPreconfirmationByDelegate { entity, signature },
+            ));
+            p2p.broadcast_preconfirmations(preconfirmations)
+                .map_err(|e| PreConfServiceError::Broadcast(format!("{e:?}")))?;
+            Ok(())
+        } else {
+            Err(PreConfServiceError::Broadcast(
+                "P2P service not available".to_string(),
+            ))
```

Comment:
I think we just should ignore then. Plus in the follow-up PR we will notify P2P and internal services(like TxStatusManager), so I think you can just return `Ok(())` always